test/remote/gateways/remote_barclaycard_smartpay_test.rb

```diff
@@ -166,7 +166,14 @@ def test_avs_result
     assert_equal 'N', response.avs_result['code']
   end
 
-  def test_nonfractional_currency
+   def test_avs_nohousenumber_result
```

Comment:
Small thing, but I'd go with `test_avs_with_no_house_number`, and generally favoring underscores over concatenating words like this, for readability. Long, explicit, descriptive names for methods and tests are fine!